templates/fine-tune-llm_v2/cookbooks/customize_initial_weights_and_prompt_format/README.ipynb

```diff
@@ -0,0 +1,203 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Customize initial weights and prompt format\n",
+    "**⏱️ Time to complete**: 60 minutes\n",
+    "\n",
+    "This guide will showcase how you can finetune a model with a similar architecture to the Llama or Mistral family and customize the chat template/ prompt format. We will focus on fine-tuning the [Meta Llama Guard 2 model](https://llama.meta.com/docs/model-cards-and-prompt-formats/meta-llama-guard-2/) throughput this cookbook. \n",
+    "\n",
+    "The two capabilities showcased here are\n",
+    "1. Bringing your own weights - (1) weights of other models similar in architecture to the Llama or Mistral family of models or (2) weights from a previous finetuning run. While we focus on (1) here, (2) is an important use-case of multi-step fine-tuning covered in depth in the cookbook [here](../continue_from_checkpoint/).\n",
+    "2. Customizing the chat template/ prompt format - Specify a custom prompt format for formatting input messages to easily fine-tune on _any_ data format.\n",
+    "\n",
+    "The Anyscale platform is uniquely suited to support both of these use-cases. This guide assumes you have familiarized yourself with the [basic fine-tuning guide](../../README.md).\n",
+    "\n",
+    "\n",
+    "# Table of Contents\n",
+    "1. [Bring your own weights](#bring-your-own-weights)\n",
+    "    - [Bring models of the same architecture](#bring-models-of-the-same-architecture)\n",
+    "        - [Example YAML](#example-yaml)\n",
+    "        - [How do I configure access to my weights in remote storage??](#)\n",
+    "            - [How do I bring my weights to Anyscale?](#how-do-I-bring-my-weights-to-Anyscale-?)\n",
+    "    - [Bring checkpoints from a previous finetuning run](#bring-checkpoints-from-a-previous-finetuning-run)\n",
+    "2. [Customizing the prompt format (chat template)](#customizing-the-prompt-format)\n",
+    "    - [How prompt formatting works in `llmforge`](#how-prompt-formatting-works-in-llmforge)\n",
+    "    - [Customizing data preprocessing and the prompt format](#customizing-data-preprocessing-and-the-prompt-format)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Bring your own weights\n",
+    "\n",
+    "In general, you can customize the initial weights in your fine-tuning run through two options in the YAML:\n",
+    "- `initial_base_model_ckpt_path` : Path to the base model weights you wish to start with\n",
+    "- `initial_adapter_model_ckpt_path`: Path to the adapter (LoRA) weights you wish to start with\n",
+    "\n",
+    "## Bring models of the same architecture\n",
+    "You can fine-tune a model similar in architecture to the Llama or Mistral family of models to fine-tune on the Anyscale Platform. For example, to fine-tune Llama Guard 2, you can \n",
```

Comment:
Suggestion: 
For example, Llama-Guard-2[link] is a model that is based on Llama-3-8B architecture that has been further finetuned on a specific dataset in a specialized format. We can tailor a model based on Llama-Guard-2 on our own dataset to improve the performance of it. The first thing we need to do is to use `meta-llama/Meta-Llama-3-8B` as the model_id but then for initial weights provide `s3://my-bucket/llama-guard-2` as the weights. We assume that the weights have been uploaded to s3 already.